R/allocate_reference_times.R

```diff
@@ -0,0 +1,337 @@
+                                        scale_factor,
+                                        prop_delay,
+                                        size_min_retro_nowcasts) {
+  sizes <- .calculate_sizes(
+    reporting_triangle,
+    max_delay,
+    scale_factor,
+    size_min_retro_nowcasts
+  )
+  n_ref_times <- sizes$n_ref_times
+  size_required <- sizes$size_required
+  size_min_delay <- sizes$size_min_delay
+  size_target <- sizes$size_target
+
+  # Check specifications against targets and requirements
+  size_used <- .check_against_requirements(
+    n_ref_times,
+    size_required,
+    size_target,
+    size_min_delay,
+    size_min_retro_nowcasts,
+    scale_factor,
+    max_delay
+  )
+
+  # If size being used equals or exceeds the target, simply split according to prop delay #nolint
+  ns <- .assign_ns_from_sizes(
+    size_used = size_used,
+    size_target = size_target,
+    size_min_delay = size_min_delay,
+    size_min_retro_nowcasts = size_min_retro_nowcasts,
+    size_required = size_required,
+    prop_delay = prop_delay
+  )
+
+  n_retrospective_nowcasts <- ns$n_retrospective_nowcasts
+  n_history_delay <- ns$n_history_delay
+
+  # Handle proportion delay warning and messages
+  .handle_output_msgs(
+    n_history_delay,
+    n_retrospective_nowcasts,
+    size_used,
+    prop_delay
+  )
+
+  return(list(
+    n_history_delay = n_history_delay,
+    n_retrospective_nowcasts = n_retrospective_nowcasts
+  ))
+}
+
+#' Handle output messages after allocation has occurred
+#'
+#' @inheritParams allocate_reference_times
+#' @param prop_delay Numeric value <1 indicating what proportion of all
+#'   reference times in the reporting triangle to be used for delay
+#'  estimation.
+#' @inheritParams .assign_ns_from_sizes
+#' @inheritParams estimate_and_apply_uncertainty
+#'
+#' @return NULL, invisibly
+.handle_output_msgs <- function(n_history_delay,
+                                n_retrospective_nowcasts,
+                                size_used,
+                                prop_delay) {
+  prop_delay_used <- n_history_delay / size_used
+
+  if (prop_delay_used != prop_delay) {
+    cli_warn(
+      message = c(
+        "`prop_delay` specified is not equivalent to `prop_delay` used.",
+        "i" = "{prop_delay} reference times were specified for delay estimation but due to minimum requirements had to be reallocated.", # nolint
+        "x" = "{round(prop_delay_used,3)} of reference times used for delay estimation." # nolint
+      )
+    )
+  }
+
+  message(sprintf(
+    "Using %d reference times for delay estimation.",
+    n_history_delay
+  ))
+  message(sprintf("Using %d reference times as retrospective nowcast times for uncertainty estimation.", n_retrospective_nowcasts)) # nolint
+  return(NULL)
+}
+
+#' Helper function to validate allocation parameters
+#'
+#'
+#' @inheritParams allocate_reference_times
+#'
+#' @returns NULL invisibly
+.validate_allocation_params <- function(scale_factor,
```

Comment:
Fix. its likely the compound if else

Reply:
I simplified the compound if else since applies to all cases